sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOrcScan.scala

```diff
@@ -1817,6 +1988,161 @@ class MultiFileCloudOrcPartitionReader(
     }
   }
 
+  private def doCombineHmbs(
+      input: Array[HostMemoryBuffersWithMetaDataBase]): HostMemoryBuffersWithMetaDataBase = {
+    val combinedMeta = computeCombinedHmbMeta(input)
+    if (combinedMeta.combinedEmptyMeta.isDefined) {
+      val ret = combinedMeta.combinedEmptyMeta.get
+      logDebug(s"Got an empty buffer after combination, number rows ${ret.numRows}")
+      ret
+    } else { // There is at least one nonempty buffer
+      // ignore the empty buffers
+      val toCombine = combinedMeta.toCombine.filterNot(_.isInstanceOf[HostMemoryEmptyMetaData])
+      logDebug(s"Using Combine mode and actually combining, number files ${toCombine.length}" +
+        s" , files: ${toCombine.map(_.partitionedFile.filePath).mkString(",")}")
+
+      val startCombineTime = System.currentTimeMillis()
+      val metaToUse = toCombine.head.asInstanceOf[HostMemoryBuffersWithMetaData]
+      val blockMetas = toCombine.flatMap(_.memBuffersAndSizes.flatMap(_.blockMeta)).toSeq
+
+      // 1) Estimate the host buffer size: header + stripes + tail (footer + postscript)
+      val combinedBufSize = estimateOutputSizeFromBlocks(blockMetas)
+
+      // 2) Allocate the buffer with the estimated size
+      val combined = closeOnExcept(HostMemoryBuffer.allocate(combinedBufSize)) { combinedBuf =>
```

Comment:
there is no chance we got this size wrong?  I think in coalescing code we double check before writing footer. but maybe that is only for parquet?

Reply:
yeah, only for parquet. According to orc coalescing code as below, there is no re-calculation for the footer size, so I think the first size should be OK. 
```
  override def calculateFinalBlocksOutputSize(
      footerOffset: Long,
      stripes: Seq[DataBlockBase],
      batchContext: BatchContext): Long = {

    // In calculateEstimatedBlocksOutputSize, we have got the true size for
    // HEADER + All STRIPES + the estimated the FileFooter size with the worst-case.
    // We return a size that is smaller than the initial size to avoid the re-allocate

    footerOffset
  }
```

Reply:
I still want the check there just in case, please add it

Reply:
Meaning before we write the footer check the sizes and reallocate the buffer if needed. The coalescing code does htis and the parquet combining code

Reply:
Done